addons/test_mail_full/tests/test_portal.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.addons.test_mail_full.tests.common import BaseFunctionalTest, TestRecipients
+from odoo.tests.common import HttpCase
+
+
+class TestPortal(BaseFunctionalTest, HttpCase, TestRecipients):
+    def setUp(self):
+        super().setUp()
+
+        self.user_1 = self.env['res.partner'].create({
```

Comment:
You could reuse partners from TestRecipients (that was the whole purpose of my comment :D ) (and this is not a user, but a partner, this may be misleading)

Reply:
I'm not sure how to proceed for that: If I use the partners in the `setUp`, I get an error saying that the partner record is not present in the table `res_partner`, and if I move my set up to `setUpClass`, it also breaks the test with an error saying that `record_portal` is missing or deleted.
Do you have any insight on this? Sorry if I'm missing something obvious.

Reply:
That's strange. They are created in setUpClass, so using them after calling setup should work. We have a lot of sub classes using those ...